broadinstitute/consent
# [DUOS-2377][risk=no] Update backend DS schema

### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2377


- Everit library is replaced with networknt JsonSchema library
- The new library has a fundamental issue with the `items` keyword in V2020-12 - however, we don't use any of the new 2020-12 functionality, so downgraded to 2019-09 without any changes needed until the library is updated.
- New dataset field: `openAccess`


Schema:
- Makes a lot of fields conditionally required in accordance to how the frontend works
- Reviewed all required fields with jonathan
- Removed some old fields
- Reworked consent group to: only allow 1 primary consent group, and only require DAC id if not open access.
- Added unit tests to ensure the schema's complex functionality works as expected


Testing
- Make calls to the create dataset endpoint, and it should give you human-readable validation messages back.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment
